gemrb/GUIScripts/LUCommon.py

```diff
@@ -37,6 +37,45 @@ def GetNextLevelExp (Level, Class):
 	# if you change it, check that all callers can handle it
 	return 0
 
+def GetNextLevels (actor, Classes):
+	"""Returns the next level the PC will attain based on current XP"""
+
+	Level = [0]*3
+
+	# Used only for TNO, all other games divide XP/Number of classes
+	XPStat = [IE_XP, IE_XP_MAGE, IE_XP_THIEF]
+
+	NumClasses = len(Classes)
+
+	for i in range(NumClasses):
+		# Get the next level we will use to look up new stats from the 2da tables
+		if GUICommon.IsNamelessOne(actor):
+			#TNO is the only character in all games that tracks individual XP stats
+			Level[i] = GetNextLevelFromExp (GemRB.GetPlayerStat (actor, XPStat[i]), Classes[i])
+		else:
+			# In all other cases, XP cut exactly by number of classes
+			Level[i] = GetNextLevelFromExp (GemRB.GetPlayerStat (actor, IE_XP)/NumClasses, Classes[i])
+
+	return Level
+
+def GetLevelDiff (actor, Level):
+	"""Calculate the number of new levels based on current levels"""
+
+	LevelDiff = [0]*3
+	LevelStat = [IE_LEVEL, IE_LEVEL2, IE_LEVEL3]
+
+	# Reverse the order of levels for dual class characters
+	# if their original class is the second class in a mc combination
+	# eg a FIGHTER that duals to MAGE will be a FIGHTER_MAGE, not a MAGE_FIGHTER
+	# so we find his Mage level in IE_LEVEL2
+	if (GUICommon.IsDualSwap(actor)):
+			LevelStat = [IE_LEVEL2, IE_LEVEL, IE_LEVEL3]
+
```

Comment:
Referring to this part here of course, I wasn't sure how the github comments would display...